/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

import "noisicaa/audioproc/public/musical_time.proto";
import "noisicaa/model/project.proto";

package noisicaa.pb;

message UpdateScoreTrack {
  required uint64 track_id = 1;
  optional int32 set_transpose_octaves = 2;
}

message UpdateScoreMeasure {
  required uint64 measure_id = 1;
  optional Clef set_clef = 2;
  optional KeySignature set_key_signature = 3;
}

message CreateNote {
  required uint64 measure_id = 1;
  optional uint32 idx = 2;
  optional Pitch pitch = 3;
  optional MusicalDuration duration = 4;
}

message UpdateNote {
  required uint64 note_id = 1;
  optional Pitch set_pitch = 2;
  optional Pitch add_pitch = 3;
  optional uint32 remove_pitch = 4;
  optional MusicalDuration set_duration = 5;
  optional uint32 set_dots = 6;
  optional uint32 set_tuplet = 7;
  message SetAccidental {
    optional uint32 pitch_idx = 1;
    optional string accidental = 2;
  }
  optional SetAccidental set_accidental = 8;
  optional int32 transpose = 9;
}

message DeleteNote {
  required uint64 note_id = 1;
}
